File: backend/src/main/java/com/aireadiness/model/Project.java
package com.aireadiness.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document(collection = "projects")
public class Project {

    @Id
    private String id;

    @Indexed
    private String userId;

    private String name;

    private String description;

    private String projectType;

    private String primaryLanguage;

    private String framework;

    private String repositoryUrl;

    private Instant createdAt = Instant.now();

    private Instant updatedAt = Instant.now();

    public Project() {
    }

    public Project(String userId, String name, String description, String projectType, String primaryLanguage, String framework, String repositoryUrl) {
        this.userId = userId;
        this.name = name;
        this.description = description;
        this.projectType = projectType;
        this.primaryLanguage = primaryLanguage;
        this.framework = framework;
        this.repositoryUrl = repositoryUrl;
        this.createdAt = Instant.now();
        this.updatedAt = Instant.now();
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getProjectType() {
        return projectType;
    }

    public void setProjectType(String projectType) {
        this.projectType = projectType;
    }

    public String getPrimaryLanguage() {
        return primaryLanguage;
    }

    public void setPrimaryLanguage(String primaryLanguage) {
        this.primaryLanguage = primaryLanguage;
    }

    public String getFramework() {
        return framework;
    }

    public void setFramework(String framework) {
        this.framework = framework;
    }

    public String getRepositoryUrl() {
        return repositoryUrl;
    }

    public void setRepositoryUrl(String repositoryUrl) {
        this.repositoryUrl = repositoryUrl;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Instant createdAt) {
        this.createdAt = createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Instant updatedAt) {
        this.updatedAt = updatedAt;
    }
}

File: frontend/src/types/project.ts
export interface Project {
  id: string;
  userId?: string;
  name: string;
  description: string;
  projectType?: string;
  primaryLanguage?: string;
  type?: string;
  language?: string;
  framework?: string;
  repositoryUrl?: string;
  createdAt?: string;
  updatedAt?: string;
  lastAnalyzed?: string;
  releasesCount?: number;
  issuesCount?: number;
  latestVersion?: string;
  latestScore?: number;
  status?: 'READY' | 'NEEDS_REVIEW' | 'NOT_READY' | 'QUEUED';
  tags?: string[];
}


export interface CreateProjectRequest {
  name: string;
  description?: string;
  projectType: string;
  primaryLanguage: string;
  framework?: string;
  repositoryUrl?: string;
}

export interface UpdateProjectRequest {
  name: string;
  description?: string;
  projectType: string;
  primaryLanguage: string;
  framework?: string;
  repositoryUrl?: string;
}
